src/tracing/perfetto/include/matter/tracing/macros_impl.h

```diff
@@ -30,3 +30,10 @@ PERFETTO_DEFINE_CATEGORIES(perfetto::Category("Matter").SetDescription("Matter t
 #define MATTER_TRACE_END(label, group) TRACE_EVENT_END("Matter")
 #define MATTER_TRACE_INSTANT(label, group) TRACE_EVENT_INSTANT("Matter", label, "class_name", group)
 #define MATTER_TRACE_SCOPE(label, group) TRACE_EVENT("Matter", label, "class_name", group)
+
+#define MATTER_TRACE_COUNTER(label, group)                                                                                         \
+    do                                                                                                                             \
+    {                                                                                                                              \
+        static int count##_label##_group = 0;                                                                                      \
+        TRACE_COUNTER("Matter", perfetto::CounterTrack(label), ++count##_label##_group);                                           \
```

Comment:
do we need CounterTrack? Docs at https://android.googlesource.com/platform/external/perfetto/+/refs/heads/master/include/perfetto/tracing/track_event.h#383 seem to say that `TRACE_COUNTER("Matter", label, ...)` would be sufficient.